### PR TITLE
Integrate Rockset as a vector store

### PR DESCRIPTION
**Pull request by gadhagod**
# Description

Added [Rockset](https://rockset.com/docs) as a vector store to llama index.

> Rockset is a real-time analytics database which enables queries on massive, semi-structured data without operational burden. Rockset is serverless and fully managed. It offloads the work of managing configuration, cluster provisioning, denormalization, and shard / index management. Rockset is also SOC 2 Type II compliant and offers encryption at rest and in flight, securing and protecting any sensitive data. Most teams can ingest data into Rockset and start executing queries in less than 15 minutes.

Added example notebook and integration tests. 

Ran `black` and `mypy` and everything's good.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Disiok** [comment]
Thanks for the contribution! If you could resolve the merge conflict, would be happy to help merge this!

**Comment by gadhagod** [comment]
Thanks @Disiok. Merge conflicts are now fixed.

File: docs/examples/vector_stores/RocksetIndexDemo.ipynb
Comment (logan-markewich) [inline docs/examples/vector_stores/RocksetIndexDemo.ipynb:1]: Do you mind adding an example showing how to load an index from an existing collection?

Should be something as simple as this, but good to remind users that this is how it works haha
```
vector_store = <setup vector store to point to existing collection>
index = VectorStoreIndex.from_vector_store(vector_store, service_context=service_context)

query_engine = index.as_query_engine()
response = query_engine.query("some test query to confirm it works")
print(str(response))
```

File: llama_index/vector_stores/rocksetdb.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        client = rockset.RocksetClient(
+            api_key=api_key or getenv("ROCKSET_API_KEY"),
+            host=api_server or getenv("ROCKSET_API_SERVER"),
+        )
+    return client
+
+
+class RocksetVectorStore(VectorStore):
+    stores_text: bool = True
+    is_embedding_query: bool = True
```
Comment (logan-markewich) [inline llama_index/vector_stores/rocksetdb.py:64]: does Rockset require flat metadata? We should add the attribute here, similar to other vector stores

**Review by logan-markewich — COMMENTED** [review]
Looks good to me overall! Just a few comments 👍🏻

**Comment by logan-markewich** [comment]
ah also, many tests are failing 👀

File: docs/examples/vector_stores/RocksetIndexDemo.ipynb
Comment (gadhagod) [reply docs/examples/vector_stores/RocksetIndexDemo.ipynb:1]: Added in https://github.com/jerryjliu/llama_index/pull/7111/commits/80c394571be5151a8eef79ee8af2978b5e754f0e.

File: llama_index/vector_stores/rocksetdb.py (hunk 1 above)
Comment (gadhagod) [reply llama_index/vector_stores/rocksetdb.py:64]: Added in https://github.com/jerryjliu/llama_index/pull/7111/commits/ee9dfcd2d637f15a2612f19876741a41dccde078.

**Comment by gadhagod** [comment]
@logan-markewich thanks for your review. I have addressed your comments in my latest changes. Please take a look.

BTW no tests are failing now.

File: docs/examples/vector_stores/RocksetIndexDemo.ipynb
Comment (logan-markewich) [reply docs/examples/vector_stores/RocksetIndexDemo.ipynb:1]: Awesome!

**Comment by logan-markewich** [comment]
Thanks for the work on this @gadhagod ! Merging!

**Pull request merged**